app/javascript/src/Stats/buyer/stats_application.jsx

```diff
@@ -16,12 +23,24 @@ export class StatsApplicationSourceCollector extends StatsSourceCollector {
   getSources (options) {
     let id = options.selectedApplicationId
     let selectedMetricName = options.selectedMetricName
-    let requestNew = (id && id !== this.id)
-    let url
+    return super.getSources({id, selectedMetricName})
+  }
+}
 
-    if (requestNew) url = this._metricUrl(id)
+export class StatsApplicationChartManager extends StatsUsageChartManager {
+  updateMetrics () {
+    let id = this.statsState.state.selectedApplicationId
+    let url = this._metricUrl(id)
+    this.sourceCollector.getMetrics(url).then(list => this.metricsSelector.update(this._groupMetrics(list)))
```

Comment:
Trying locally, I'm getting this error when reaching this line:
`Uncaught TypeError: Cannot read property 'then' of undefined`

Reply:
That's weird, `getMetrics`returns a Promise. Can you make sure you are not pulling from cache?

Reply:
I don't see this error, probably npm did not install or webpack did no compile.

Reply:
I reinstalled everything, the error has gone, maybe an npm issue...